### PR TITLE
[skip ci] Nimbus testbed for nfs volume nightly test

### LINKED ISSUE #7642: Nightly-6.0-5-22-NFS-Volume - Nimbus Pod is unavailable
```Suite setup failed:
Keyword 'Setup ESX And NFS Suite' failed after retrying 10 times. The last error was: '2018-04-03 23:42:52 -0700: VM resource requirement: cpu - 1 VCPU(s), mem - 1024 MB, disk - 41943040 KB
2018-04-03 23:42:52 -0700: /mts/git/nimbus/lib/nimbus/deploys/nfs/nfs_deploy.rb, "NFS-292099.490703"
2018-04-03 23:42:52 -0700: sc-prd-rdops-nimbusgateway001 pid 56042
2018-04-03 23:42:52 -0700: USER: svc.victestuser
2018-04-03 23:42:52 -0700: id: uid=2000540(svc.victestuser) gid=201 groups=201

2018-04-03 23:42:52 -0700: logname: svc.victestuser

2018-04-03 23:42:52 -0700: Git version for /mts/git/: heads/staging-0-g5dc9da5
2018-04-03 23:42:52 -0700: Git version for /mts/git/nimbus: heads/staging-0-g33f2ee5
2018-04-03 23:42:52 -0700: Git version for /mts/git/nimbus-configs: heads/staging-0-g8dafca4
2018-04-03 23:42:53 -0700: Git version for /mts/git/rbvmomi: heads/staging-0-g142d4a3
2018-04-03 23:42:53 -0700: Git version for /mts/git/rbvmomi-utils: heads/master-0-gad4be2b
    [ Message content over the limit has been removed. ]
    "/mts/git/nimbus-toolchain/lin64/ruby/2.2.0/lib/ruby/2.2.0/timeout.rb:34:in `block in catch'",
    "/mts/git/nimbus-toolchain/lin64/ruby/2.2.0/lib/ruby/2.2.0/timeout.rb:34:in `catch'",
    "/mts/git/nimbus-toolchain/lin64/ruby/2.2.0/lib/ruby/2.2.0/timeout.rb:34:in `catch'",
    "/mts/git/nimbus-toolchain/lin64/ruby/2.2.0/lib/ruby/2.2.0/timeout.rb:104:in `timeout'",
    "/mts/git/nimbus-toolchain/lin64/ruby/2.2.0/lib/ruby/2.2.0/timeout.rb:125:in `timeout'",
    "/mts/git/nimbus/lib/nimbus/base_deploy.rb:622:in `run_cli'",
    "/mts/git/bin/../nimbus/lib/nimbus/deploys/nfs/nfs_deploy.rb:225:in `<main>'"
  ]
}
--- end nimbus json results ---
2018-04-03 23:42:53 -0700: Terminating VIM connection ...
2018-04-03 23:42:53 -0700: Exception: NimbusExceptionUnavailablePod: Pod is unavailable - Specified pod ["sc-prd-vc049"] is busy and cannot take any new load.' does not contain 'To manage this VM use'```

### PR DESCRIPTION
**Pull request by rajanashok**
Updated the NFS Volume nightlies to build a full NFS test bed instead of relying on the unsupported pod placement calls.

Fixes #7642

### CODE REVIEW

File: tests/manual-test-cases/Group5-Functional-Tests/5-22-NFS-Volume.robot [hunk 1]
```diff
@@ -38,23 +38,14 @@ Setup ESX And NFS Suite
...
 
-    ${esx1}  ${esx1_ip}=  Deploy Nimbus ESXi Server  %{NIMBUS_USER}  %{NIMBUS_PASSWORD}
-    Open Connection  %{NIMBUS_GW}
-    Wait Until Keyword Succeeds  2 min  30 sec  Login  %{NIMBUS_USER}  %{NIMBUS_PASSWORD}
-    ${POD}=  Fetch POD  ${esx1}
-    Log To Console  ${POD}
-    Close Connection
-
-    ${nfs}  ${nfs_ip}=  Deploy Nimbus NFS Datastore  %{NIMBUS_USER}  %{NIMBUS_PASSWORD}  additional-args=--nimbus ${POD}
+    ${nfs}  ${nfs_ro}  ${esx1}  ${nfs_ip}  ${nfs_ro_ip}  ${esx1_ip}=  Deploy Nimbus NFS Datastore  %{NIMBUS_USER}  %{NIMBUS_PASSWORD}  additional-args=--testbedSpecRubyFile /dbc/w3-dbc302/rashok/vic-nfs.rb --esxBuild ${ESX_VERSION} --esxPxeDir ${ESX_VERSION} --plugin testng
```
Comment (mhagen-vmware) [inline tests/manual-test-cases/Group5-Functional-Tests/5-22-NFS-Volume.robot]: just make sure that path is accessible by the nightly user account

File: tests/resources/nimbus-testbeds/vic-nfs.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+require '/mts/git/nimbus/lib/testframeworks/testng/testng.rb'
+oneGB = 1 * 1000 * 1000
+$testbed = Proc.new do |sharedStorageStyle, esxStyle|
+  esxStyle ||= 'fullInstall'
+  esxStyle = esxStyle.to_s
+  sharedStorageStyle = sharedStorageStyle.to_s
+
+  testbed = {
+    'version' => 3,
+    'name' => "vsan-1esx-allflash-#{sharedStorageStyle}-#{esxStyle}",
```
Comment (mhagen-vmware) [inline tests/resources/nimbus-testbeds/vic-nfs.rb]: it is not allflash, it apparently has one HDD that is 250GB in size?

File: tests/resources/nimbus-testbeds/vic-nfs.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    'esx' => (0..0).map do |i|
```
Comment (mhagen-vmware) [inline tests/resources/nimbus-testbeds/vic-nfs.rb]: this is a single iteration loop, really not needed

File: tests/resources/nimbus-testbeds/vic-nfs.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    'name' => 'dev-nfs',
+    'type' => 'NFS',
+    'nfsOpt' => 'rw',
+    'numMem' => 64 * 1024,
+    },
+    'nfs' => {
+    'name' => 'dev-nfs',
+    'type' => 'NFS',
+    'nfsOpt' => 'ro',
+    'numMem' => 64 * 1024,
```
Comment (mhagen-vmware) [inline tests/resources/nimbus-testbeds/vic-nfs.rb]: I like the memory increase, might be overkill but is there any way to specify SSD for these?  and we should specify a good sized disk capacity in addition to the mem update

File: tests/resources/nimbus-testbeds/vic-nfs.rb [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        'name' => "esx.#{i}",
+        'style' => esxStyle,
+        'numMem' => 64 * 1024,
+        'numCPUs' => 6,
+        'disks' => [ 250 * oneGB ],
+        'disableNfsMounts' => true,
+        'ssds' => [ 200 * oneGB, 250 * oneGB ],
+        'nics' => 2,
+        'staf' => false,
+        'desiredPassword' => 'ca$hc0w',
```
Comment (mhagen-vmware) [inline tests/resources/nimbus-testbeds/vic-nfs.rb]: e2eFunctionalTest

File: tests/resources/nimbus-testbeds/vic-nfs.rb [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    'name' => "vsan-1esx-allflash-#{sharedStorageStyle}-#{esxStyle}",
+    'esx' => (0..0).map do |i|
+      {
+        'name' => "esx.#{i}",
+        'style' => esxStyle,
+        'numMem' => 64 * 1024,
+        'numCPUs' => 6,
+        'disks' => [ 250 * oneGB ],
+        'disableNfsMounts' => true,
+        'ssds' => [ 200 * oneGB, 250 * oneGB ],
```
Comment (mhagen-vmware) [inline tests/resources/nimbus-testbeds/vic-nfs.rb]: two of these are really not needed I don't think.

File: tests/resources/nimbus-testbeds/vic-nfs.rb [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        'nics' => 2,
+        'staf' => false,
+        'desiredPassword' => 'ca$hc0w',
+        'vmotionNics' => ['vmk0'],
+        'post_postboot_commands' => [
+            'esxcli system settings kernel set -s autoPartition -v "TRUE"', \
+            'esxcli system settings kernel set -s skipPartitioningSsds -v "TRUE"', \
+            'services.sh restart', \
+            'esxcli vsan storage tag add -d mpx.vmhba1:C0:T1:L0 -t capacityFlash', \
+            'esxcli storage core adapter rescan --all', \
```
Comment (mhagen-vmware) [inline tests/resources/nimbus-testbeds/vic-nfs.rb]: we don't need vsan here

File: tests/resources/nimbus-testbeds/vic-nfs.rb (hunk 5 above)
Comment (rajanashok) [reply tests/resources/nimbus-testbeds/vic-nfs.rb]: I am updating the password to e2eFunctionalTest after the ESX is created. https://github.com/rajanashok/vic/blob/15813ff2ec286414a36201f381079212ca9b55bc/tests/resources/Nimbus-Util.robot#L445

File: tests/resources/nimbus-testbeds/vic-nfs.rb [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    'numMem' => 16 * 1024,
+    },
+    'nfs' => {
+    'name' => 'dev-nfs',
+    'type' => 'NFS',
+    'nfsOpt' => 'ro',
+    'numMem' => 16 * 1024,
+    }],
+  }
+  testbed
```
Comment (mhagen-vmware) [inline tests/resources/nimbus-testbeds/vic-nfs.rb]: stray comment?

File: tests/resources/nimbus-testbeds/vic-nfs.rb [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+        'style' => esxStyle,
+        'numMem' => 16 * 1024,
+        'numCPUs' => 6,
+        "disks" => [ 30 * oneGB, 30 * oneGB, 30 * oneGB],
+        'disableNfsMounts' => true,
+        'nics' => 2,
+        'staf' => false,
+        'desiredPassword' => 'e2eFunctionalTest',
+    }],
+    'vsan' => true,
```
Comment (mhagen-vmware) [inline tests/resources/nimbus-testbeds/vic-nfs.rb]: un-needed

File: tests/resources/Nimbus-Util.robot [hunk 10]
```diff
@@ -405,25 +405,39 @@ Get Vsphere Version
...
+    \   ${status}=  Run Keyword And Return Status  Should Contain  ${line}  ${name}.nfs.1' is up. IP:
+    \   ${ip}=  Run Keyword If  ${status}  Fetch From Right  ${line}  ${SPACE}
+    \   Run Keyword If  ${status}  Set Suite Variable  ${nfs-ro-ip}  ${ip}
+    \   Exit For Loop If  ${status}
+
+    :FOR  ${line}  IN  @{out}
+    \   ${status}=  Run Keyword And Return Status  Should Contain  ${line}  ${name}.esx.0' is up. IP:
+    \   ${ip}=  Run Keyword If  ${status}  Fetch From Right  ${line}  ${SPACE}
+    \   Run Keyword If  ${status}  Set Suite Variable  ${esx-ip}  ${ip}
+    \   Exit For Loop If  ${status}
```
Comment (mhagen-vmware) [inline tests/resources/Nimbus-Util.robot]: these loops should be replaced with the 'Get IP' logic we are using in newer test cases like this:
https://github.com/vmware/vic/blob/master/tests/manual-test-cases/Group5-Functional-Tests/5-27-Selenium-Grid.robot#L33

File: tests/resources/Nimbus-Util.robot [hunk 11]
```diff
@@ -405,25 +405,39 @@ Get Vsphere Version
 Deploy Nimbus NFS Datastore
```
Comment (mhagen-vmware) [inline tests/resources/Nimbus-Util.robot]: Deploy Simple NFS Testbed

File: tests/resources/Nimbus-Util.robot (hunk 10 above)
Comment (rajanashok) [reply tests/resources/Nimbus-Util.robot]: Yup thanks, these are pretty handy! Updated the PR.

**Pull request merged**